### DISCUSSION TITLE
Preferences - Categories - Restrictions by user

### DISCUSSION THREAD

**Discussion by MihailOpanasenko** [magicbell/community#151] [Questions] [answered]
Is there a way to show preferences to users with alerts in these categories? 

For example:
I have a user that needs to show only messages from the A and B categories 
another user should see only the message from the C and D categories 

Receiving a message works fine but they see all of the settings for all the categories

I am using buld-in code to display messages

**Comment by smeijer** [comment]
Hi,

Are you using the react sdk to show your user's inbox and preferences? If you do, it's possible to customize the preferences shown:

```tsx
import { MagicBellProvider, NotificationPreferences } from '@magicbell/magicbell-react';

function Component() {
  return (
    <MagicBellProvider apiKey={API_KEY} userEmail={USER_EMAIL}>
      <NotificationPreferences channels={['email']} categories={['announcements', 'billing']} />
    </MagicBellProvider>
  );
}
```

Please see here for the docs: https://www.npmjs.com/package/@magicbell/magicbell-react#notificationpreferences

In this example, I'm rendering the preferences component standalone so it can be embedded on say a user settings page. But it's also possible to provide it to the Inbox using the [NotificationPreferences](https://www.npmjs.com/package/@magicbell/magicbell-react#notificationinbox) property, to replace the default checkboxes with your own implementation.

**Reply by MihailOpanasenko** [reply]
No, I am using an embedded snippet, Is there are option that allows to specify used categories?

**Reply by smeijer** [reply]
Sorry, I'm afraid we don't support any custom elements, or config on this level, for the embeddable.

**Reply by smeijer** [reply, marked as answer]
I think if you don't use react, the best option at this moment would be to use our [js api client](https://www.npmjs.com/magicbell) and build your own preferences component.
